src/Socket/FileSocket.php

```diff
@@ -21,13 +21,13 @@ abstract class FileSocket implements SocketInterface
      */
     private $socket;
 
-    protected function __construct(mixed $socket, Timeout $receiveTimeout)
+    protected function __construct(mixed $socket, protected Timeout $receiveTimeout)
```

Comment:
```suggestion
    protected function __construct(mixed $socket, private Timeout $receiveTimeout)
```

I think you can promote `$socket` too, and if I'm right, both can be readonly.